File: src/co/uk/taurasystems/application/ui/tabpanes/job/JobTab.java
package co.uk.taurasystems.application.ui.tabpanes.job;

import java.io.IOException;

import co.uk.taurasystems.db.models.Job;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.control.Tab;
import javafx.scene.layout.AnchorPane;

public class JobTab extends Tab {
	
	private Job job;

	public JobTab(Job job) {
		this.job = job;
		setText("ID: "+job.getID());
		load();
	}
	
	public void load() {
		FXMLLoader fxmlLoader = new FXMLLoader(JobTabPaneController.class.getResource("JobTabPane.fxml"));
		try {
			Parent tabContents = (Parent)fxmlLoader.load();
			JobTabPaneController controller = (JobTabPaneController)fxmlLoader.getController();
			controller.setJob(job);
			controller.initialize(fxmlLoader.getLocation(), fxmlLoader.getResources());
			AnchorPane pane = new AnchorPane();
			AnchorPane.setTopAnchor(tabContents, 0.0);
			AnchorPane.setRightAnchor(tabContents, 0.0);
			AnchorPane.setLeftAnchor(tabContents, 0.0);
			AnchorPane.setBottomAnchor(tabContents, 0.0);
			pane.getChildren().addAll(tabContents);
			setContent(pane);
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
	
	public Job getJob() {
		return job;
	}
}

File: src/co/uk/taurasystems/application/ui/tabpanes/job/JobTab.kt
package co.uk.taurasystems.application.ui.tabpanes.job

import java.io.IOException

import co.uk.taurasystems.db.models.Job
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.control.Tab
import javafx.scene.layout.AnchorPane

class JobTab(val job: Job) : Tab() {

    init {
        text = "ID: " + job.ID
        load()
    }

    fun load() {
        val fxmlLoader = FXMLLoader(JobTabPaneController::class.java.getResource("JobTabPane.fxml"))
        try {
            val tabContents = fxmlLoader.load<Any>() as Parent
            val controller = fxmlLoader.getController<Any>() as JobTabPaneController
            controller.setJob(job)
            controller.initialize(fxmlLoader.location, fxmlLoader.resources)
            val pane = AnchorPane()
            AnchorPane.setTopAnchor(tabContents, 0.0)
            AnchorPane.setRightAnchor(tabContents, 0.0)
            AnchorPane.setLeftAnchor(tabContents, 0.0)
            AnchorPane.setBottomAnchor(tabContents, 0.0)
            pane.children.addAll(tabContents)
            content = pane
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}
